app/models/piece.rb

```diff
@@ -36,7 +44,11 @@ def captured?
   def valid_move?(to_row, to_col)
     return false if move_nil?(to_row, to_col)
     return false if move_out_of_bounds?(to_row, to_col)
-    return false if move_destination_same_side?(to_row, to_col)
+    return false if move_destination_ally?(to_row, to_col)
+
+    # until have Knight::move_legal?
+    return true if type == "Knight"
```

Comment:
You can override `move_obstructed?` in the Knight class so that it simply returns `true` when calling `knight_piece.move_obstructed?`.